src/world/area_kkj/kkj_25/B06D10.c

```diff
@@ -1,35 +1,176 @@
+    unkStruct->scale.z = SPRITE_WORLD_SCALE;
+    unkStruct->foldID = func_8013A704(1);
+    unkStruct->entityID = create_generic_entity_world(NULL, kkj_25_UnkFoldFunc);
+
+    evt_set_variable(script, GW(10), (s32) unkStruct);
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240800_B07200(Evt* script, s32 isInitialCall) {
+    UnkEntityStruct* unkStruct = (UnkEntityStruct*) evt_get_variable(NULL, GW(10));
+
+    func_8013A854(unkStruct->foldID);
+    free_generic_entity(unkStruct->entityID);
+    heap_free(unkStruct);
+    evt_set_variable(script, GW(10), 0);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_8024086C_B0726C(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 x = evt_get_float_variable(script, *args++);
+    s32 y = evt_get_float_variable(script, *args++);
+    s32 z = evt_get_float_variable(script, *args++);
+    UnkEntityStruct* unkStruct = (UnkEntityStruct*) evt_get_variable(NULL, GW(10));
+
+    unkStruct->pos.x = x;
+    unkStruct->pos.y = y;
+    unkStruct->pos.z = z;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_8024092C_B0732C(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 x = evt_get_float_variable(script, *args++);
+    s32 y = evt_get_float_variable(script, *args++);
+    s32 z = evt_get_float_variable(script, *args++);
+    UnkEntityStruct* unkStruct = (UnkEntityStruct*) evt_get_variable(NULL, GW(10));
+
+    unkStruct->rot.x = x;
+    unkStruct->rot.y = y;
+    unkStruct->rot.z = z;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802409EC_B073EC(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 x = evt_get_float_variable(script, *args++);
+    s32 y = evt_get_float_variable(script, *args++);
+    s32 z = evt_get_float_variable(script, *args++);
+    UnkEntityStruct* unkStruct = (UnkEntityStruct*) evt_get_variable(NULL, GW(10));
+
+    unkStruct->scale.x = x;
+    unkStruct->scale.y = y;
+    unkStruct->scale.z = z;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240AAC_B074AC(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 spriteIndex = evt_get_variable(script, *args++);
+    s32 rasterIndex = evt_get_variable(script, *args++);
+    UnkEntityStruct* unkStruct = (UnkEntityStruct*) evt_get_variable(NULL, GW(10));
+
+    unkStruct->spriteIndex = spriteIndex;
+    unkStruct->rasterIndex = rasterIndex;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240B18_B07518(Evt* script, s32 isInitialCall) {
+    EffectInstance* effect;
+
+    fx_stars_orbiting(0, script->varTable[0], script->varTable[1], script->varTable[2], script->varTable[3], script->varTable[4], &effect);
+    script->varTablePtr[0] = effect;
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240B8C_B0758C(Evt* script, s32 isInitialCall) {
+    EffectInstance* effect = (EffectInstance*) script->varTablePtr[0];
+
+    effect->data.starsOrbiting->unk_28 = 0;
```

Comment:
Assumed the data to be a `StarsOrbitingFXData` since we have some calls to `fx_stars_orbiting` in this file. Was this the right call or should I revert these changes?

Reply:
Lgtm